README.md

```diff
@@ -1 +1,173 @@
-# vertx.rest
\ No newline at end of file
+# vertx-rest
+[![Continuous Integration](https://github.com/dream11/vertx-rest/actions/workflows/ci.yml/badge.svg)](https://github.com/dream11/rest/actions/workflows/ci.yml)
+[![Code Coverage](https://codecov.io/gh/dream11/vertx-rest/branch/master/graph/badge.svg)](https://codecov.io/gh/dream11/vertx-rest)
+![License](https://img.shields.io/badge/license-MIT-green.svg)
+
+- [Overview](#overview)
+- [How to use](#usage)
+  - [Rest Verticle](#create-rest-verticle)
+    - [Injection](#injection)
+  - [JAX-RS Routes](#create-a-rest-resource)
+    - [Validations](#validations)
+    - [Exception Handling](#exception-handling)
+    - [Swagger](#create-swagger-specificationui)
+    - [Timeouts](#timeouts)
+    - [Example Usage](#example-usage)
+- [References](#references)
+
+## Overview
+The vert-rest library aims to make it simple to create a REST application using vertx. A number of Request and Response Filters, Exception Mappers and features like back-pressure, validations, timeout, logging are provided.
+
+It internally uses [resteasy-vertx](https://github.com/resteasy/resteasy/tree/main/server-adapters/resteasy-vertx)
+## Usage
+
+Add the following dependency to the *dependencies* section of your build descriptor:
+
+- Maven (in your `pom.xml`):
+```xml
+  <dependency>
+    <groupId>com.dream11</groupId>
+    <artifactId>vertx-rest</artifactId>
+    <version>LATEST</version>
```

Comment:
Do not add `LATEST` here as it is deprecated in maven 3. Instead add x.y.z which is hyperlink to release page